домашнее задание урока 2.2. ООП.

andr777 opened the pull request.

andr777 committed:
домашнее задание урока 2.2. ООП.

dmitry-bizin commented on src/pro/sky/java/homeworks/course2/lesson1/Bicycle.java:
```diff
@@ -0,0 +1,13 @@
+
+public class Bicycle implements vehicle{
+
+    public String modelName;
```
нет инкапсуляции

dmitry-bizin commented on src/pro/sky/java/homeworks/course2/lesson1/Car.java:
```diff
@@ -0,0 +1,18 @@
+
+public class Car implements vehicle {
+
+    public String modelName;
```
нет инкапсуляции

dmitry-bizin commented on src/pro/sky/java/homeworks/course2/lesson1/ServiceStation.java:
```diff
@@ -0,0 +1,59 @@
+    */
+
+
+    // полиморфизм
```
это не полиморфизм)) это перегрузка, а нужно сделать полиморфизм.

Чтобы был полиморфизм - создай интерфейс с одним методом проверки транспорта. Все 3 типа транспорта пусть его реализуют, но каждый по-своему. В классе ServiceStation создаёшь метод, который принимает этот новый интерфейс как параметр. Внутри метода одна строка - вызов метода интерфейса. В main создаёшь ТС и передаёшь сначала один, потом второй и третий - это и будет полиморфизм

dmitry-bizin commented on src/pro/sky/java/homeworks/course2/lesson1/Truck.java:
```diff
@@ -0,0 +1,19 @@
+
+public class Truck  implements vehicle {
+
+    public String modelName;
```
нет инкапсуляции

dmitry-bizin commented on src/pro/sky/java/homeworks/course2/lesson1/vehicle.java:
```diff
@@ -0,0 +1,10 @@
+package pro.sky.java.homeworks.course2.lesson1;
+
+public interface vehicle {
```
названия интерфейсов и классов с большой буквы

andr777 committed:
домашнее задание урока 2.2. ООП.
переделал, теперь к обектам обращение через интерфейс Vehicle,
переменные в классах private

BizinMitya commented on src/pro/sky/java/homeworks/course2/lesson1/Bicycle.java:
```diff
@@ -0,0 +1,33 @@
+
+public class Bicycle implements Vehicle {
+
+    private String modelName;
```
вот эти общие поля нужно вынести в базовый класс-родитель. Они у тебя дублируются во всех классах

dmitry-bizin commented on src/pro/sky/java/homeworks/course2/lesson1/Car.java:
```diff
@@ -0,0 +1,42 @@
+
+public class Car implements Vehicle {
+
+    private String modelName;
```
вот эти общие поля нужно вынести в базовый класс-родитель. Они у тебя дублируются во всех классах

dmitry-bizin commented on src/pro/sky/java/homeworks/course2/lesson1/Truck.java:
```diff
@@ -0,0 +1,45 @@
+
+public class Truck implements Vehicle {
+
+    private String modelName;
```
вот эти общие поля нужно вынести в базовый класс-родитель. Они у тебя дублируются во всех классах

dmitry-bizin commented on src/pro/sky/java/homeworks/course2/lesson1/Vehicle.java:
```diff
@@ -0,0 +1,24 @@
+
+public interface Vehicle {
+
+    default void updateTyre() {
```
не, здесь слишком много методов. Должен быть один метод всего в этом интерфейсе

andr777 committed:
домашнее задание урока 2.2. ООП.
переделал, теперь к обектам обращение через интерфейс Vehicle,
переменные в классах private,
добавил extends, неиспользуемые методы удалил
